rekuho/src/lib/Portion.svelte

```diff
@@ -0,0 +1,177 @@
+  const pxToPortionPosition = ([x, y]: Position): Position => {
+    const div = (x: number, y: number): number => Math.floor(x / y)
+    const column = div(x - view_rect.x, column_width)
+    const line = div(y - view_rect.y, line_height)
+    return [column, line]
+  }
+
+  const font = theme.font
+  const line_height: number = 18
+  const column_width: number = getCharacterWidth(`${font.weight} ${font.size} ${font.family}`) || 1
+
+  // TODO: Implement proper selections
+  let selection: Position | null = null
+
+  const mousedown = (ev: MouseEvent) => {
+    selection = pxToPortionPosition([ev.pageX, ev.pageY])
+    cursorUpdate(0, selection)
+    input.focus()
+    ev.preventDefault()
+  }
+
+  const mouseup = (ev: MouseEvent) => {
+    if (selection) {
+      const begin = selection
+      selection = null
+      const end = pxToPortionPosition([ev.pageX, ev.pageY])
+    }
+  }
+
+  const mousemove = (ev: MouseEvent) => {
+    if (selection) {
+      cursorUpdate(0, pxToPortionPosition([ev.pageX, ev.pageY]))
+    }
+  }
+
+  const keydown = (ev: KeyboardEvent) => {
+    // TODO: Handle input from keydown events properly
+    ev.preventDefault() // stops the textarea from filling up with input
+    if (ev.key.length === 1 && isAlpha(ev.key.charCodeAt(0))) {
+      insertAt(ev.key, $cursors[0].pos)
+      cursorMove(0, [1, 0])
+    }
+  }
+
+  const gutter_mousedown = (line: number, ev: MouseEvent) => {
+    cursorUpdate(0, [null, line])
+  }
+</script>
+
+<div class="view" bind:this={view} style:width="{width}px" style:height="{height}px">
+  <!--<GutterColumn line_height={line_height} />-->
+  <!-- TODO: Maybe place into ex. GutterColumn.svelte -->
+  <div
+    class="gutter"
+    style:background={theme.gutter.background}
+    style:width="{gutter_width}px"
+    style:height="{height}px">
+    {#each $lines as _, i}
+      <div
+        class="gutter-cell"
+        on:mousedown={(e) => {
+          gutter_mousedown(i, e)
+        }}
+        style:font-size={theme.font.size}
+        style:height="{line_height}px"
+        style:top="{i * line_height}px">
+        {i + 1}
+      </div>
+    {/each}
+  </div>
+  <div
+    style:position="absolute"
+    style:height="{height}px"
+    style:top="0"
+    style:background={theme.editor_background}
+    style:width="{theme.text_offset}px"
+    style:left="{gutter_width}px" />
+  <div
+    bind:this={container}
+    class="container"
+    on:mousedown={mousedown}
+    on:mousemove={mousemove}
+    on:mouseup={mouseup}
+    style:background={theme.editor_background}
+    style:left="{gutter_width + theme.text_offset}px">
+    <textarea
+      bind:this={input}
+      wrap="off"
+      tabindex="-1"
+      on:keydown={keydown}
+      style:user-select="text"
+      style:position="absolute"
+      style:width="{column_width}px" />
+    <LinesView bind:theme {line_height} />
+    <CursorsLayer bind:theme {column_width} {line_height} />
+  </div>
```

Comment:
Styles like this definitely do not scale

Reply:
So to clarify, the problem I see here is that, if we want to have most of the UI elements be configurable (i.e. gutter padding, line-height...) we are going to end up with files that contain hundreds and hundreds of lines of inlined styles, intertwined with the HTML. 

We do not need to use inline styles for anything other than dynamic properties, that is, properties that will change on certain conditions or events. 

For configurable styles (!= dynamic), we can just provide some kind of abstraction over a global, external CSS config.

There is also https://svelte.dev/blog/svelte-css-in-js. 
> It's important to note that most CSS-in-JS libraries have a runtime library, and many don't support statically extracting styles out into a separate .css file at build time (which is essential for the best performance). You should therefore only use CSS-in-JS if it's necessary for your application!